--- a/arch/arm64/boot/dts/mediatek/mt7988a-bananapi-bpi-r4-pro.dtsi
+++ b/arch/arm64/boot/dts/mediatek/mt7988a-bananapi-bpi-r4-pro.dtsi
@@ -370,7 +370,7 @@

 /* M.2 (11280000) 1L0 key-m SSD1 CN13 / key-b SIM1 CN15 */
 &pcie2 {
-	status = "okay";
+	status = "disabled";
 };

 /* M.2 (11290000) 1L1 key-m SSD2 CN14 / key-b SIM2 CN18 */
@@ -437,7 +437,7 @@
 	pcie-2-hog {
 		gpio-hog;
 		gpios = <79 GPIO_ACTIVE_HIGH>;
-		output-high;
+		output-low;
 	};

 	/* 1L1 0=key-b (CN18), 1=key-m (CN14) */
